src/ripple/app/misc/ValidatorSite.h

```diff
@@ -73,10 +74,33 @@ class ValidatorSite
         {
             clock_type::time_point refreshed;
             ListDisposition disposition;
+            std::string message;
         };
 
-        std::string uri;
-        parsedURL pUrl;
+        struct Resource
+        {
+            explicit Resource(std::string u);
+            std::string uri;
+            parsedURL pUrl;
+        };
+        using ResourcePtr = std::shared_ptr<Resource>;
```

Comment:
I’d prefer moving away from this style and just using `std::shared_ptr<Resource>`. These aliases just end up obscuring things.